src/components/ExportCapture.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useContext } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Checkbox,
+  FormControl,
+  FormGroup,
+  FormControlLabel,
+} from '@material-ui/core';
+import { CapturesContext } from '../context/CapturesContext';
+import { CSVLink } from "react-csv";
+import { formatCell } from './CaptureTable';
+import { tokenizationStates } from 'common/variables';
+import { getDateTimeStringLocale } from 'common/locale';
+
+const useStyle = makeStyles((theme) => ({
+  container: {
+    position: 'relative',
+    padding: theme.spacing(0, 4),
+  },
+  textInput: {
+    margin: theme.spacing(2, 1),
+    flexGrow: 1,
+  },
+  formControl: {
+    margin: theme.spacing(3),
+  },
+  csvLink: {
+    textDecoration: "none",
+    color: theme.palette.stats.green,
+  }
+}));
+
+const ExportCapture = (props) => {
+  const { isOpen, handleClose, columns, filter, speciesState } = props;
+  const classes = useStyle();
+  let nameColumns = {};
+  columns.forEach(({ attr }) => {
+    nameColumns[attr] = true;
+  });
+  const [checkedColumns, setCheckColumns] = useState(nameColumns);
+  const capturesContext = useContext(CapturesContext);
+  const [downloadData, setDownloadData] = useState([]);
+
+  function handleChange(attr) {
+    setCheckColumns({ ...checkedColumns, [attr]: false });
+  }
+
+  function processDownloadData(captures, selectedColumns) {
```

Comment:
There's a lot of unnecessary duplication in this function that will make it more difficult to maintain and error prone:

- Use the `renderer` values that are already defined in `columns` rather than defining them all again.
- If you must define new renderers for any reason, make that the only thing you set inside the `if - else` statement. Only include the `formatCapture[attr] = formatCell(...` statement once.